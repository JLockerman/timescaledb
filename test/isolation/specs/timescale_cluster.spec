setup
{
 CREATE TABLE ts_cluster_test(time timestamptz, temp float, location int);
 SELECT create_hypertable('ts_cluster_test', 'time', chunk_time_interval => interval '1 day');
 INSERT INTO ts_cluster_test VALUES ('2017-01-20T09:00:01', 23.4, 1),
       ('2017-01-21T09:00:01', 21.3, 2),
       ('2017-01-22T09:00:01', 19.5, 3);
}

teardown { DROP TABLE ts_cluster_test; }

session "s1"
setup		{ BEGIN; }
step "s1a"	{ SELECT * FROM ts_cluster_test; }
step "s1b"	{ INSERT INTO ts_cluster_test VALUES ('2017-01-22T09:00:01', 19.5, 3); }
step "s1c"	{ COMMIT; }

session "s2"
step "s2"	{ CLUSTER ts_cluster_test USING ts_cluster_test_time_idx; }

permutation "s1a" "s2" "s1b" "s1c"
