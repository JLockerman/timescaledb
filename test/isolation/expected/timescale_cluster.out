Parsed test spec with 2 sessions

starting permutation: s1a s2 s1b s1c
step s1a: SELECT * FROM ts_cluster_test;
time           temp           location       

Fri Jan 20 09:00:01 2017 PST23.4           1              
Sat Jan 21 09:00:01 2017 PST21.3           2              
Sun Jan 22 09:00:01 2017 PST19.5           3              
step s2: CLUSTER ts_cluster_test USING ts_cluster_test_time_idx; <waiting ...>
step s1b: INSERT INTO ts_cluster_test VALUES ('2017-01-22T09:00:01', 19.5, 3);
step s2: <... completed>
error in steps s1b s2: ERROR:  deadlock detected
step s1c: COMMIT;
